pg_diffix/aggregation/common.h

```diff
@@ -164,7 +166,7 @@ typedef struct Bucket
 struct AnonAggFuncs
 {
   /* Get type information of final value. */
-  void (*final_type)(Oid *type, int32 *typmod, Oid *collid);
+  void (*final_type)(Oid primary_arg_type, Oid *type, int32 *typmod, Oid *collid);
```

Comment:
ok, given your "ah crap" comment I guess this one is obsolete. I think the additional plumbing would not be worth it.

Reply:
Long term I think it's worth it instead of bloating the function signature. I don't think it's difficult to get the args down or even recompute them on the fly.

Reply:
Let me have a look where that function is called.

Reply:
`build_args_desc` can be called with only an `Aggref` and the result requires a single `pfree`. So it's cheap to compute on the fly rather than drill down.